lib/drivers/medtronic/medtronicDriver.js

```diff
@@ -1120,15 +1134,53 @@ module.exports = function (config) {
             if(err) {
               return cb(err,null);
             }
-            callback(null, result);
+
+            // pump only stores 36 history pages
+            var MAX_HISTORY_PAGES = 36;
+            var lastPosition = null;
+            var currentPosition = result.currentPage;
+
+            if (_.has(cfg, 'lastUpload.client.private.delta')) {
+              // retrieve last position from most recent upload record
+              lastPosition = cfg.lastUpload.client.private.delta.lastPosition;
+            }
+
+            _.merge(cfg, { delta : {lastPosition : currentPosition}}); // will form part of upload record
+
+            var currentPage = positionToPage(lastPosition, currentPosition, 'pump history');
+            callback(null, { currentPage : Math.min(currentPage, MAX_HISTORY_PAGES) + 1 });
           });
         },
         current_cbg_page : function(callback){
           getOneRecord(readCurrentCBGPage(), false, true, function(err, result) {
             if(err) {
               return cb(err,null);
             }
-            callback(null, result);
+
+            // pump stores a maximum of 32 glucose and ISIG history pages
+            var MAX_CBG_PAGES = 32;
+            var lastCBGPosition = null;
+            var lastGlucosePosition = null;
+            var lastIsigPosition = null;
+
+            if (_.has(cfg, 'lastUpload.client.private.delta')) {
+              // retrieve last positions from most recent upload record
+              lastCBGPosition = cfg.lastUpload.client.private.delta.lastCBGPosition;
+              lastGlucosePosition = cfg.lastUpload.client.private.delta.lastGlucosePosition;
+              lastIsigPosition = cfg.lastUpload.client.private.delta.lastIsigPosition;
+            }
+
+            _.merge(cfg, { delta : {
+              lastCBGPosition : result.currentCBGPosition,
+              lastGlucosePosition : result.currentGlucosePosition,
+              lastIsigPosition : result.currentIsigPosition
+            }}); // will form part of upload record
+
+            var currentPage = positionToPage(lastCBGPosition, result.currentCBGPosition, 'CBG history');
+            var glucosePage = positionToPage(lastGlucosePosition, result.currentGlucosePosition, 'glucose');
+            var isigPage = positionToPage(lastIsigPosition, result.currentIsigPosition, 'isig');
+
+            callback(null, { currentPage : currentPage, glucosePage: Math.min(glucosePage, MAX_CBG_PAGES) + 1, isigPage: Math.min(isigPage, MAX_CBG_PAGES) + 1 });
```

Comment:
The original code doesn't `+1` to the result of the `Math.min` function, unless this is an additional bug fix.

Reply:
Good catch, but yes, it's an additional bug fix. Basically it's so that we always retrieve the most recent page, even if we already retrieved it in the past, as there may be new records on that page.

Reply:
I wonder if this is what is causing the issue we're seeing. I'm removing the `+1` here so that it's the same as the original code again.